src/raftstore/store/util.rs

```diff
@@ -578,19 +578,31 @@ pub fn conf_state_from_region(region: &metapb::Region) -> ConfState {
     conf_state
 }
 
-pub struct KeysInfoFormatter<'a>(pub &'a [Vec<u8>]);
-
-impl<'a> fmt::Display for KeysInfoFormatter<'a> {
+pub struct KeysInfoFormatter<
+    'a,
+    I: std::iter::DoubleEndedIterator<Item = &'a Vec<u8>>
+        + std::iter::ExactSizeIterator<Item = &'a Vec<u8>>
+        + Clone,
+>(pub I);
+
+impl<
+        'a,
+        I: std::iter::DoubleEndedIterator<Item = &'a Vec<u8>>
+            + std::iter::ExactSizeIterator<Item = &'a Vec<u8>>
+            + Clone,
+    > fmt::Display for KeysInfoFormatter<'a, I>
+{
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self.0.len() {
-            0 => write!(f, "no key"),
-            1 => write!(f, "key \"{}\"", escape(self.0.first().unwrap())),
+        let mut it = self.0.clone();
```

Comment:
Why clone?

Reply:
Since `iter.next() / next_back()` mutates itself, while `fmt()` only accept `&self`. Anyway this `clone()` only clones the reference iterator itself, which should be pretty cheap.